src/lightning/fabric/strategies/xla.py

```diff
@@ -84,6 +84,17 @@ def _configure_launcher(self) -> None:
         self._launcher = _XLALauncher(self)
 
     def setup_environment(self) -> None:
+        from torch_xla.experimental.pjrt import using_pjrt
+
+        assert self.parallel_devices is not None
+        if using_pjrt() and len(self.parallel_devices) == 1:
```

Comment:
Added this to `setup_environment` since it's the first strategy hook to get called after the strategy has been lazy initialized.

Reply:
The user could create the strategy themselves, and the connector call `strategy.parallel_devices = ...` as part of the lazy initialization logic

Reply:
I could move this check to the parallel devices setter, if that's better

Reply:
In this case, having it here is fine with me too. You could still consider removing `the using_pjrt()` since even in the case with xrt, it was never the intention to have XLAStrategy being used for a single process execution.

Reply:
It could be useful for debugging, just as we support ddp_spawn with 1 device. In fact I suspect that the "one core" test you commented below used it for this reason. And XRT did seem to support it.